client/lib/i18n-utils/switch-locale.js

```diff
@@ -286,12 +287,15 @@ function addRequireChunkTranslationsHandler( localeSlug = i18n.getLocaleSlug(),
 			return;
 		}
 
-		const translationChunkPromise = getTranslationChunkFile( chunkId, localeSlug ).then(
-			( translations ) => {
+		const translationChunkPromise = getTranslationChunkFile( chunkId, localeSlug )
+			.then( ( translations ) => {
 				addTranslations( translations, userTranslations );
 				loadedTranslationChunks[ chunkId ] = true;
-			}
-		);
+			} )
+			.catch( ( error ) => {
+				debug( `Encountered an error loading translation chunk ${ chunkId }.` );
+				debug( error );
```

Comment:
It would be great if we could report to Sentry that we failed to load a translation chunk. Including info on the chunk name and locale. That would be an improvement over the current `TypeError: Failed to fetch`. And would help us figure out what is causing these failures.

Calypso client already integrates the `@sentry/browser` library. Maybe @fullofcaffeine or @noahtallen know how to report custom errors there.

Reply:
Yes! You can use `import { captureException } from 'calypso/lib/sentry';`, and then call `captureException(error)`. (You could create a `new Error()` to include extra information.)

It's important to import this function from our internal lib instead of from the sentry package to avoid loading extra code when it's not needed. Our lib makes sure sentry is only async loaded for a small percent of requests, but lets you use these stub functions anywhere, even if they are sometimes no-op when sentry isn't loaded.

Reply:
Thank you for the suggestion! I've updated the PR to report the error to Sentry.

However, I was unable to test it locally (I believe Sentry is not enabled in local environment). @noahtallen, do you happen to know how to test if the error reporting works as expected?

Reply:
I don't think it would be blocking to test the error flow, but I think you could test it locally by commenting out these lines of code:

https://github.com/Automattic/wp-calypso/blob/d6c65e414a1041b53f2efc726d4bf664e0484718/client/lib/sentry/index.ts#L135-L148

And then somehow triggering the exception. I think it will show up in Sentry at that point

Reply:
Thanks - I managed to to test it on calypso.live and it looks like errors are now properly reported to Sentry. I'll proceed with merging the PR.